staking-bank/src/lib.rs

```diff
@@ -0,0 +1,510 @@
+#[receive(
+    contract = "staking_bank",
+    name = "validators",
+    parameter = "AccountAddress",
+    return_value = "(AccountAddress,String)"
+)]
+fn validators<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    _host: &impl HasHost<State, StateApiType = S>,
+) -> ReceiveResult<(AccountAddress, String)> {
+    let id: AccountAddress = ctx.parameter_cursor().get()?;
+
+    match id {
+        VALIDATOR_0 => Ok((id, "https://validator.umb.network".to_string())),
+        VALIDATOR_1 => Ok((id, "https://validator2.umb.network".to_string())),
+        _ => bail!(CustomContractError::NotValidator.into()),
+    }
+}
+
+// Sandbox constants and functions
+
+#[cfg(feature = "sandbox")]
+const VALIDATOR_0: AccountAddress = AccountAddress([0u8; 32]);
+#[cfg(feature = "sandbox")]
+const VALIDATOR_1: AccountAddress = AccountAddress([1u8; 32]);
+
+/// The number of validators.
+#[cfg(feature = "sandbox")]
+const NUMBER_OF_VALIDATORS: u8 = 2;
+/// total supply = number_of_validators * ONE.
+#[cfg(feature = "sandbox")]
+const TOTAL_SUPPLY: u64 = 2 * 1000000000000000000u64;
+
+#[cfg(feature = "sandbox")]
+/// Internal function that returns a boolean if the given address is a validator.
+fn _is_validator(_validator: AccountAddress) -> bool {
+    _validator == VALIDATOR_0 || _validator == VALIDATOR_1
+}
+
+#[cfg(feature = "sandbox")]
+/// Internal function that returns all validators.
+fn _addresses() -> Vec<AccountAddress> {
+    vec![VALIDATOR_0, VALIDATOR_1]
+}
+
+/// View function that returns validator's URL (as well as the inputted account address). The function throws an error if the address is not a validator.
+#[cfg(feature = "sandbox")]
+#[receive(
+    contract = "staking_bank",
+    name = "validators",
+    parameter = "AccountAddress",
+    return_value = "(AccountAddress,String)"
+)]
+fn validators<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    _host: &impl HasHost<State, StateApiType = S>,
+) -> ReceiveResult<(AccountAddress, String)> {
+    let id: AccountAddress = ctx.parameter_cursor().get()?;
+
+    match id {
+        VALIDATOR_0 => Ok((id, "https://validator.umb.network".to_string())),
+        VALIDATOR_1 => Ok((id, "https://validator2.umb.network".to_string())),
+        _ => bail!(CustomContractError::NotValidator.into()),
+    }
+}
+
+#[derive(Serial, Deserial)]
+pub struct State {}
+
+/// All smart contract errors.
+#[derive(Debug, PartialEq, Eq, Reject, Serial, SchemaType)]
+enum CustomContractError {
+    /// Failed to parse the parameter.
+    #[from(ParseError)]
+    ParseParams, // -1
+    /// Failed to log because the log is full.
+    LogFull, // -2
+    /// Failed to log because the log is malformed.
+    LogMalformed, // -3
+    /// Failed to invoke a contract.
+    InvokeContractError, // -4
+    /// Failed because the validators count is not consistent.
+    #[allow(dead_code)]
+    ValidatorsCountMisMatch, // -5
+    /// Failed because the address is not a validator.
+    #[allow(dead_code)]
```

Comment:
Because these two error codes are only used when the contract is compiled with one of the three features `development`, `production`, or `sandbox`. Clippy complains about unused code here.